components/brave_wallet/renderer/js_solana_provider.cc

```diff
@@ -68,6 +70,10 @@ JSSolanaProvider::JSSolanaProvider(content::RenderFrame* render_frame)
     *g_provider_solana_web3_script =
         LoadDataResource(IDR_BRAVE_WALLET_SOLANA_WEB3_JS);
   }
+  if (g_wallet_standard_script->empty()) {
+    *g_wallet_standard_script = LoadDataResource(
```

Comment:
Do we really need to cache `LoadDataResource()` returned values?
If we need, then it should be shared memory for all the renderer. Otherwise it takes 1.8MB(the string size) per renderer process because it's private process memory.

Reply:
For large resources:
1. If the resource isn't gzipped use `GetRawDataResource()` each time, don't cache;
2. If unzip time is affordable don't cache, use `LoadDataResourceString()` each time (except large resources on UI thread in browser process);
3. Store cached unzipped resources in browser process. Pass them via IPC each time you need it. Or store it in shared memory and pass a handle.

Bad practice:
* Using private process memory to cache `LoadDataResourceString()` (i.e. static std::string) for large resources. It increases total footprint by `RESOURCE_SIZE * N` where `N` is renderer process number. Also each new tab usually uses a new process => old cache doesn't help to save time.
* Loading large zipped resources on UI thread. It will result in freezes/frame drops: crbug.com/973417

Reply:
In this particular case all the resources are unzipped => please use `GetRawDataResource()` each time, don't cache.

P.S. I wonder could we support gzip in [transpile_web_ui](https://github.com/brave/brave-core/blob/enable-brave-features-for-perf/components/common/typescript.gni#L83).

Reply:
Those cached scripts resources are removed and they are only be used once, I don't think they need to be cached.